### DISCUSSION TITLE
Strange startup issue of the server

### DISCUSSION THREAD

**Discussion by mdziczkowski** [rabbitmq/rabbitmq-server#6000] [Other] [answered]
### Test environment:

**OS**:

1. Ubuntu
2. Debian

**OS versions**:

ad. 1. : Xenial and later (upgrade)
ad 2. : Buster and Bookworm

**RabbitMQ version**: avaliable from the OS repository (installed from the first tested OS version and then upgraded)

### Problem description:

I had created the `rabbitmq.config` file from the scratch using the documentation avaliable on the rabbitmq website and I had used the sysctl format for it.

When attempting to start the server, the server has returned a following error message:
```
* Starting RabbitMQ Messaging Server rabbitmq-server
* FAILED - check /var/log/rabbitmq/startup_\{log, _err\}
```
Content of the `startup_log` file:

```
{"could not start kernel pid",application_controller,"error in config file \"/etc/rabbitmq/rabbitmq.config\" (none): no ending <dot> found"}
```

**Comment by michaelklishin** [comment, marked as answer]
If you use the new format then the name of the file [must be `rabbitmq.conf`](https://www.rabbitmq.com/configure.html#config-file-formats). `advanced.config` and `rabbitmq.config` use the Erlang term format.

While not related, Ubuntu and Debian are usually a good year behind [latest RabbitMQ releases](https://www.rabbitmq.com/changelog.html), although Bookworm right now [happens to be an exception and ships `3.10.8`](https://packages.debian.org/search?keywords=rabbitmq-server&searchon=names&suite=all&section=all).

**Reply by mdziczkowski** [reply]
@michaelklishin I need to correct you. Acording to the RabbitMQ documentation, the `rabbitmq.conf` is described as a "New style format (sysctl or ini-like)" and the `rabbitmq.advanced.config` is as Erlang one

**Reply by lukebakken** [reply]
> Acording to the RabbitMQ documentation, the rabbitmq.config is described as a "New style format (sysctl or ini-like)"

Incorrect. See the source code to the website here - https://github.com/rabbitmq/rabbitmq-website/blob/live/site/configure.md?plain=1#L264-L265

> and the rabbitmq.advanced.config is as Erlang one

Incorrect - https://github.com/rabbitmq/rabbitmq-website/blob/live/site/configure.md?plain=1#L273-L274

**Comment by mdziczkowski** [comment]
I had used following site: https://rabbitmq.com/configure.html as reference, where The "clasic" Erlang version is not too well described.

I would like to propose one extension for the config files, with would accept both styles. This will prevent confusion in the future

**Reply by michaelklishin** [reply]
We are not interested in this idea. First of all, the extension tells the server how to load the file.
Second of all, supporting any extension and any format will *create more* confusion.

The separation is very clear right now: `.config` (including `rabbitmq.config` in the classic format) means the Erlang term format, and nodes still can load the original `rabbitmq.config`. `rabbitmq.conf` for the sysctl-style format and `advanced.config` as an optional Erlang-term file for what cannot be expressed in the main one.

**Reply by michaelklishin** [reply]
`rabbitmq.config` is the original filename and is only supported for backwards compatibility. That's why we do not really cover it in much detail in the docs.

The absolute majority of users prefer the new format, it is easier to generate, read, harder to run into a syntax error, and so on.
